plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -272,16 +287,52 @@ public function register_checkout_field( $options ) {
 			trigger_error( sprintf( 'Registering a field with hidden set to true is not supported. The field "%s" will be registered as visible.', esc_html( $id ) ), E_USER_WARNING );
 		}
 
-		// Insert new field into the correct location array.
-		$this->additional_fields[ $id ] = array(
+		$field_data = array(
 			'label'          => $options['label'],
 			'hidden'         => false,
+			'type'           => $type,
 			'optionalLabel'  => empty( $options['optionalLabel'] ) ? '' : $options['optionalLabel'],
 			'required'       => empty( $options['required'] ) ? false : $options['required'],
 			'autocomplete'   => empty( $options['autocomplete'] ) ? '' : $options['autocomplete'],
 			'autocapitalize' => empty( $options['autocapitalize'] ) ? '' : $options['autocapitalize'],
 		);
 
+		if ( 'select' === $type ) {
+			if ( empty( $options['options'] ) || ! is_array( $options['options'] ) ) {
+				return new \WP_Error( 'woocommerce_blocks_checkout_select_field_no_options_specified', __( 'Fields of type "select" must have an array of "options".', 'woocommerce' ) );
+			}
+
+			$cleaned_options = array();
+			$added_values    = array();
+
+			// Check all entries in $options['options'] has a key and value member.
+			foreach ( $options['options'] as $key => $option ) {
+				if ( ! isset( $option['value'] ) || ! isset( $option['label'] ) ) {
+					return new \WP_Error( 'woocommerce_blocks_checkout_select_field_options_invalid', __( 'Fields of type "select" must have an array of "options" with a "value" and "label" member.', 'woocommerce' ) );
+				}
+
+				$sanitized_value = sanitize_text_field( $option['value'] );
+				$sanitized_label = sanitize_text_field( $option['label'] );
+
+				if ( in_array( $sanitized_value, $added_values, true ) ) {
+					// translators: %s is the duplicate value.
+					return new \WP_Error( 'woocommerce_blocks_checkout_select_field_options_not_unique', sprintf( __( 'The value in each option of "select" fields must be unique. Duplicate value "%s" found.', 'woocommerce' ), esc_html( $sanitized_value ) ) );
```

Comment:
What are your thoughts on adding line breaks for readability-reasons?

```suggestion
					return new \WP_Error(
						'woocommerce_blocks_checkout_select_field_options_not_unique',
						sprintf(
							__( 'The value in each option of "select" fields must be unique. Duplicate value "%s" found.', 'woocommerce' ),
							esc_html( $sanitized_value )
						)
					);
```